### ISSUE TITLE
Support for ReactNative for Windows

### ISSUE THREAD

**Issue by FearMediocrity** [realm/realm-js#432]
After speaking with Tim Oliver, it was suggested I post here a request for support of React Native on Windows Mobile.

While Windows support for ReactNative is still in development, it is coming soon, and was worthy an announcement at the Facebook F8 conference.

It would be nice to see Realm as a part of this expansion.

**Comment by alazier** [comment]
This is definitely on our radar

**Comment by adisoftbn** [comment]
+1
Might be nice to have it, since you need some persistent data storage.

**Comment by juhasuni** [comment]
+1
Any news about this topic? React Native Windows is definitely missing Realm.

**Comment by Steviey** [comment]
+1

**Comment by asteinbps** [comment]
+1

**Comment by luxn** [comment]
+1

**Comment by metalmalte** [comment]
+1

**Comment by kenji4569** [comment]
+1

**Comment by csotiriou** [comment]
+1

**Comment by leequarella** [comment]
+1

**Comment by rozele** [comment]
+1, and I'd love to help out with these. I actually [just tweeted](https://twitter.com/EricRozell/status/897930327167049728) about this - would love to chat further.

**Comment by bmunkholm** [comment]
@rozele Feel free to reach out to me at bm@realm.io for a chat.

**Comment by amorenew** [comment]
+1

**Comment by Traviskn** [comment]
Would the existing realm-dotnet repository be of any use for supporting react-native for windows?  I imagine much of the C# code will be similar

**Comment by juhasuni** [comment]
Any news on this topic?

**Comment by jakabjack** [comment]
@Traviskn i don't think so. i'm just struggling to install realm for react native windows. I managed to use realm in a C# UWP app, using the realm nuget package. I was also able to put that package in the react-native-windows generated .sln and use it from within the launching project from c#. But it's not working from javascript side.
Whenever android react native links realm, it adds a line into MainApplication.java: 
`    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new RealmReactPackage()
      );
    }
`
besides of course adding the realm dependency to gradle.
I was thinking something similar has to be done for the react-native-windows project as well:
`    class MainReactNativeHost : ReactNativeHost
    {
        public override string MainComponentName => "testPrj";

#if !BUNDLE || DEBUG
        public override bool UseDeveloperSupport => true;
#else
        public override bool UseDeveloperSupport => false;
#endif

        protected override string JavaScriptMainModuleName => "index";

#if BUNDLE
        protected override string JavaScriptBundleFile => "ms-appx:///ReactAssets/index.windows.bundle";
#endif

        protected override List<IReactPackage> Packages => new List<IReactPackage>
        {
            new MainReactPackage(),
        };
    }
`
As you can see it has the same structure as the android app, and some line would be needed below `new MainReactPackage(),` for `new RealmReactPackage()`.
Unfortunately the Realm .NET nuget package does not contain such an object.

**Comment by Udbhav12** [comment]
Any update on this ?

**Comment by hakimny** [comment]
Any update on when realm will have support for react-native-windows?

**Comment by creambyemute** [comment]
@bmunkholm Did your chat with @rozele have any outcomes?

**Comment by bmunkholm** [comment]
@creambyemute Nope - I never heard anything.

**Comment by juhasuni** [comment]
@alazier, any chance of moving this forward? MS has put recently great effort on renewing the React Native for Windows and it would be great to be able to use Realm as the database for the Windows apps built on React Native. @rozele do you know if there are any plans on your side?

**Comment by elcssmouhsine** [comment]
anyone managed to make Realm work with react-native-windows ?

**Comment by SerhiiRubelLightIT** [comment]
react-native-windows definitely needs Realm support

**Comment by juhasuni** [comment]
Would be great to see this working with react-native-windows

**Comment by marianolc** [comment]
+1 for this. This would be a great feature.

**Comment by xuanhuan95** [comment]
+1 for this

**Comment by wecliang** [comment]
+1 for this

**Comment by raaannzzz** [comment]
+1 for this

**Comment by lkhore** [comment]
+120000

**Comment by lkhore** [comment]
+1 for this

**Comment by BMecoli** [comment]
+1 for this

**Comment by kneth** [comment]
It is a bit early to say much. We would like to replace our current RN integration with our new JSI/Hermes integration and see if [JSI/V8](https://github.com/microsoft/v8-jsi) is a way forward. So many things to consider before we can give a definite answer.

**Comment by alanBriceno** [comment]
Any news on this topic?

**Comment by Murthrag** [comment]
+1

**Comment by kiranchenna** [comment]
is there any update regarding support for react native windows. we would love to use realm for our development. we are still missing it.

hope we will hear good news soon.

**Comment by kneth** [comment]
It is not on our short-term roadmap but we follow the development closely.

**Comment by homaily** [comment]
+1
